proposals/mips/mip-b06/mip-b06.sol

```diff
@@ -0,0 +1,310 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.19;
+
+import "@forge-std/Test.sol";
+
+import {MToken} from "@protocol/MToken.sol";
+import {Configs} from "@proposals/Configs.sol";
+import {Proposal} from "@proposals/proposalTypes/Proposal.sol";
+import {Addresses} from "@proposals/Addresses.sol";
+import {JumpRateModel} from "@protocol/IRModels/JumpRateModel.sol";
+import {TimelockProposal} from "@proposals/proposalTypes/TimelockProposal.sol";
+import {CrossChainProposal} from "@proposals/proposalTypes/CrossChainProposal.sol";
+import {Comptroller} from "@protocol/Comptroller.sol";
+
+contract mipb06 is Proposal, CrossChainProposal, Configs {
+    string public constant name = "MIP-b06";
+    uint256 public constant timestampsPerYear = 60 * 60 * 24 * 365;
+    uint256 public constant SCALE = 1e18;
+
+    uint256 public constant USDC_PREVIOUS_CF = 0.8e18;
+    uint256 public constant USDC_NEW_CF = 0.82e18;
+
+    uint256 public constant WETH_PREVIOUS_CF = 0.78e18;
+    uint256 public constant WETH_NEW_CF = 0.8e18;
+
+    struct IRParams {
+        uint256 kink;
+        uint256 baseRatePerTimestamp;
+        uint256 multiplierPerTimestamp;
+        uint256 jumpMultiplierPerTimestamp;
+    }
+
+    constructor() {
+        bytes memory proposalDescription = abi.encodePacked(
+            vm.readFile("./proposals/mips/mip-b06/MIP-B06.md")
+        );
+        _setProposalDescription(proposalDescription);
+    }
+
+    function _validateJRM(address jrmAddress, address tokenAddress, IRParams memory params)
+        internal
+    {
+        JumpRateModel jrm = JumpRateModel(jrmAddress);
+        assertEq(
+            address(MToken(tokenAddress).interestRateModel()),
+            address(jrm),
+            "interest rate model not set correctly"
+        );
+
+        assertEq(jrm.kink(), params.kink, "kink verification failed");
+        assertEq(
+            jrm.timestampsPerYear(),
+            timestampsPerYear,
+            "timestamps per year verifiacation failed"
+        );
+        assertEq(
+            jrm.baseRatePerTimestamp(),
+            (params.baseRatePerTimestamp * SCALE) / timestampsPerYear / SCALE,
+            "base rate per timestamp validation failed"
+        );
+        assertEq(
+            jrm.multiplierPerTimestamp(),
+            (params.multiplierPerTimestamp * SCALE) / timestampsPerYear / SCALE,
+            "multiplier per timestamp validation failed"
+        );
+        assertEq(
+            jrm.jumpMultiplierPerTimestamp(),
+            (params.jumpMultiplierPerTimestamp * SCALE) /
+                timestampsPerYear /
+                SCALE,
+            "jump multiplier per timestamp validation failed"
+        );
```

Comment:
there's a lot of duplicate code here from MIP-B05, it would be great to refactor this out into a fixture / helper file so that this logic doesn't need to be duplicated